lib/elixir/lib/module.ex

```diff
@@ -1246,16 +1256,17 @@ defmodule Module do
     end
   end
 
-  defp compile_doc(table, line, kind, pair, args, doc, env) do
+  defp compile_doc(table, line, kind, pair, args, doc, env, impl) do
     signature = build_signature(args, env)
 
     case :ets.lookup(table, {:doc, pair}) do
       [] ->
+        doc = if impl, do: false, else: doc
```

Comment:
What happens if you do set `impl` true and you do set a doc though? The doc should still be there. Maybe we need a test for this too?

Reply:
I think we need something like:

```elixir
doc = if is_nil(doc) && impl, do: false, else: doc
```